File: scripts/Geo.Coding.py
import sys
import os
import re
import datetime

sys.path.append('/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages')
import requests

for item in current_selected_items:
	#print item.getProperties()
	for prop in item.getProperties().entrySet():
		if prop.key == "homeStreetName":
			homeStreetName = prop.value
		if prop.key == 'homeState':
			homeState = prop.value
		if prop.key == 'homeCity':
			homeCity = prop.value
		if prop.key == 'homeZip':
			homeZip = prop.value
	print homeStreetName + ' ' + homeCity + ' ' + homeState + ' ' + homeZip

	payload = {'sentence': itemText}
	try:
		# https://github.com/artpar/languagecrunch/
		response = requests.get('http://localhost:8080/nlp/parse', params=payload)
		data = response.json()

File: scripts/geo-coding.rb
require 'net/http'
require 'uri'
require 'json'
require 'java'

def geolocate(address, apiKey, item)
	uri = URI.parse('https://maps.googleapis.com/maps/api/geocode/json?address=' + address + '&key=' + apiKey)
	https = Net::HTTP.new(uri.host, uri.port)
	https.use_ssl=true
	begin
		request = Net::HTTP::Get.new(uri.request_uri)
		response = https.request(request)
		payload = JSON.parse(response.body)
		results = payload["results"]
		status = payload["status"]
		
		if status != "OK"
			return "status=#{status}"
		end
		custom_metadata = item.getCustomMetadata
		formattedAddress = results[0]["formatted_address"] #todo: update to support multiple locations
		lat = results[0]["geometry"]["location"]["lat"]
		custom_metadata.putFloat("Latitude",lat)
		lng = results[0]["geometry"]["location"]["lng"]
		custom_metadata.putFloat("Longitude",lng)
		out = Hash.new
		out["address"] = formattedAddress
		out["lat"] = lat
		out["lng"]= lng
		return out
	rescue => ex
		puts "ERROR: #{ex.message}"
	end
end


#address = '4108+Hain+Drive+Lafayette+Hill,PA+19444'
key = 'AIzaSyBCond6OMDJ5dO-ypZ8Zbn6OFJXnx8wQQM'

currentSelectedItems.each do |item|
	homeStreetName = item.getProperties['homeStreetName']
	homeCity = item.getProperties['homeCity']
	homeState = item.getProperties['homeState']
	homeZip = item.getProperties['homeZip']
	completeAddress = "#{homeStreetName}, #{homeCity}, #{homeState} #{homeZip}"
	puts completeAddress.gsub(' ','+')
	puts geolocate(completeAddress, key, item)
end

